cmd/kusk/cmd/kubectl.go

```diff
@@ -0,0 +1,135 @@
+/*
+The MIT License (MIT)
+
+Copyright © 2022 Kubeshop
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+.
+*/
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/component-base/logs"
+	kubectl "k8s.io/kubectl/pkg/cmd"
+)
+
+// type CmdOpts config.CLIOptions
+
+type kubectlPluginHandler struct{}
+
+func (h *kubectlPluginHandler) Lookup(filename string) (string, bool) {
+	path, err := exec.LookPath(fmt.Sprintf("kubectl-%s", filename))
+	if err != nil || path == "" {
+		return "", false
+	}
+	return path, true
+}
+
+// adapted from kubectl.DefaultPluginHandler
+func (h *kubectlPluginHandler) Execute(executablePath string, cmdArgs, environment []string) error {
+	if _, err := exec.LookPath("kubectl"); err != nil {
+		if exe, err := os.Executable(); err == nil {
+			logrus.Warn("kubectl not found in $PATH. many kubectl plugins try to run 'kubectl'.`", exe)
+		}
+	}
+
+	// Windows does not support exec syscall.
+	if runtime.GOOS == "windows" {
```

Comment:
Is this intentional, because reading the documentation at <https://pkg.go.dev/runtime>

> GOOS is the running program's operating system target: one of darwin, freebsd, linux, and so on. To view possible combinations of GOOS and GOARCH, run "go tool dist list".

Running that command, I see:

```sh
$ go tool dist list | grep -i windows
windows/386
windows/amd64
windows/arm
windows/arm6
```

Maybe a regular expression match might be better.

Reply:
runtime.GOOS doesn't return architecture 
```
func main() {
	fmt.Println(runtime.GOOS)
}
```
```
% go run main.go
darwin
```